dj/models/annotation.py

```diff
@@ -0,0 +1,83 @@
+"""
+Models for annotations.
+"""
+from typing import TYPE_CHECKING, List, Optional
+
+from sqlalchemy import JSON, String, UniqueConstraint
+from sqlalchemy.sql.schema import Column as SqlaColumn
+from sqlmodel import Field, Relationship
+
+from dj.models.base import BaseSQLModel
+from dj.models.node import NodeType
+from dj.typing import ColumnType
+
+if TYPE_CHECKING:
+    from dj.models import Column
+
+
+RESERVED_ANNOTATION_NAMESPACE = "system"
+
+
+class AnnotationTypeName(BaseSQLModel):
+    """
+    Annotation type name.
+    """
+
+    name: str = Field(sa_column=SqlaColumn("name", String))
+
+
+class AnnotationTypeBase(AnnotationTypeName):
+    """
+    Available annotation types for column metadata (base model).
+    """
+
+    namespace: str
+    description: str
+    uniqueness_scope: List[str] = Field(sa_column=SqlaColumn(JSON))
```

Comment:
This could probably be some sort of enum made of Column, Node, System, whatever else we want to have these attributes/annotations on, right?

Reply:
Yeah, makes sense! I added an enum and also some checks based on that.